src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -78,7 +78,7 @@ public HybridQueryBuilder(StreamInput in) throws IOException {
     protected void doWriteTo(StreamOutput out) throws IOException {
         writeQueries(out, queries);
         if (isClusterOnOrAfterMinReqVersionForPaginationInHybridQuery()) {
-            out.writeInt(paginationDepth);
+            out.writeOptionalInt(paginationDepth);
```

Comment:
I ran into a similar error as this with the StreamOutput on k-NN.  Using `out.getVersion()` to set what value to write might help avoid null checks elsewhere. StreamInput has a similar method to so you can set a default for the value.

Example: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/indices/ModelMetadata.java#L83

Reply:
I will address this in https://github.com/opensearch-project/neural-search/issues/1097